yt_dlp/extractor/yandexvideo.py

```diff
@@ -268,35 +270,75 @@ class ZenYandexChannelIE(InfoExtractor):
         },
         'playlist_mincount': 657,
     }]
 
-    def _entries(self, id, url):
-        webpage = self._download_webpage(url, id)
-        data_json = self._parse_json(re.findall(r'var\s?data\s?=\s?({.+?})\s?;', webpage)[-1], id)
-        for key in data_json.keys():
-            if key.startswith('__serverState__'):
-                data_json = data_json[key]
-        items = list(try_get(data_json, lambda x: x['feed']['items'], dict).values())
-        more = try_get(data_json, lambda x: x['links']['more']) or None
+    def _entries(self, channel_title, server_state_json, server_settings_json):
+        items = list(try_get(server_state_json, lambda x: x['feed']['items'], dict).values())
+
+        if(len(items) == 0):
+            items = list(try_get(server_settings_json, lambda x: x['exportData']['items']))
+
+        prev_next_page_id = None
+        more = try_get(
+            server_state_json, lambda x: x['links']['more']
+        ) or try_get(
+            server_settings_json, lambda x: x['exportData']['more']['link']) or None
+
+        if more:
+            next_page_id = try_get(parse_qs(more), lambda x: x['next_page_id'][0]) or None
+        else:
+            next_page_id = None
+
         for page in itertools.count(1):
-            for item in items:
+            for item in filter(lambda x: x.get('type') == 'gif', items):
                 video_id = item.get('publication_id') or item.get('publicationId')
                 video_url = item.get('link')
                 yield self.url_result(video_url, ie=ZenYandexIE.ie_key(), video_id=video_id.split(':')[-1])
-            if not more:
+            if (
+                not more
+                or len(items) == 0
+                or (next_page_id and next_page_id == prev_next_page_id)
+                or not next_page_id
+            ):
                 break
-            data_json = self._download_json(more, id, note='Downloading Page %d' % page)
+            data_json = self._download_json(more, channel_title, note='Downloading Page %d' % page)
             items = data_json.get('items', [])
             more = try_get(data_json, lambda x: x['more']['link']) or None
 
+            prev_next_page_id = next_page_id
+            if more:
+                next_page_id = try_get(parse_qs(more), lambda x: x['next_page_id'][0]) or None
+            else:
+                next_page_id = None
+
     def _real_extract(self, url):
         id = self._match_id(url)
-        return self.playlist_result(self._entries(id, url), playlist_id=id)
+        webpage = self._download_webpage(url, id)
+        data_json = self._parse_json(re.findall(r'var\s?data\s?=\s?({.+?})\s?;', webpage)[-1], id)
+        server_state_json = None
+        server_settings_json = None
+        for key in data_json.keys():
+            if key.startswith('__serverState__'):
+                server_state_json = data_json[key]
+            if key.startswith('__serverSettings__'):
```

Comment:
is there a reason u are using startswith? How does the website handles this?

doing this is kinda inefficient especially for bulk downloads.

Reply:
if there's no way to know the exact key then also add a check 

if server_state_json and server_settings_json:
  break
  
  not ideal but better than nothing

Reply:
Yes, the key name is variable but it always starts with that prefix. I agree on the break condition though, will add it. I'll wait for you to comment on the multi-line if condition from above before I do another push.